app/controllers/users.js

```diff
@@ -75,3 +79,22 @@ exports.userList = async (req, res, next) => {
     next(err);
   }
 };
+
+exports.singUpAdmin = async (req, res, next) => {
+  try {
+    const user = parseUser(req.body);
+    const result = await User.getUserBy({
+      email: user.email
+    });
+
+    if (!result) {
+      await User.createUser(user);
+    }
+    user.roleUser = roleUser.ADMINISTRATOR;
+    await User.createAdmin(user);
```

Comment:
`createAdmin` makes a update, seems tricky, also check [upsert](https://sequelize.readthedocs.io/en/latest/api/model/#upsertvalues-options-promisecreated) method in model, has a perfect fit here (create or update something)